eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -361,10 +361,9 @@ object Transactions {
 
   def makeClaimHtlcSuccessTx(commitTx: Transaction, outputs: CommitmentOutputs, localDustLimit: Satoshi, localHtlcPubkey: PublicKey, remoteHtlcPubkey: PublicKey, remoteRevocationPubkey: PublicKey, localFinalScriptPubKey: ByteVector, htlc: UpdateAddHtlc, feeratePerKw: Long): ClaimHtlcSuccessTx = {
     val redeemScript = htlcOffered(remoteHtlcPubkey, localHtlcPubkey, remoteRevocationPubkey, ripemd160(htlc.paymentHash.bytes))
-    val Some(outputIndex) = outputs.zipWithIndex.find {
-      case (CommitmentOutputLink(_, _, OutHtlc(OutgoingHtlc(outgoingHtlc))), _) => outgoingHtlc.id == htlc.id
-      case _ => false
-    }.map(_._2)
+    val Some(outputIndex) = outputs.zipWithIndex.collectFirst {
```

Comment:
Just realized that this is another "hidden cast". It may be ok here, but we need to be really careful with this pattern, this is equivalent to doing an `Option.get`, and breaks the functional model by throwing exceptions.

Reply:
I also raised this in a previous comment :).
But as a matter of fact the previous code also threw an exception here, hidden in the `findHtlc...` method. And the upper layer does a `Try` so at least we're not regressing.
But we should definitely update that function to return an `Option` in a later PR.

Reply:
Also note that the current code used to get the output index (`findPubKeyScriptIndex`) throws an exception as well.

Reply:
True, but at least `findPubKeyScriptIndex` throws an explicit `OutputNotFound`. I hesitated with proposing to throw this same exception here, but it's not *exactly* the same thing.

Reply:
Why is it not the same thing? Both method try to get the outputIndex searching the `outputs` array, if none can be found an exception will be thrown. We can replace the extractor pattern with a `.getOrElse(throw OutputNotFound)` to make it more consistent with the previous behavior. WDYT?

Reply:
I think it would be a better idea to keep it like that and change this function to return an `Option` in the refactoring PR we talked about.

Reply:
> Why is it not the same thing?

It's not the same lookup

> I think it would be a better idea to keep it like that

Agreed, that's why I acked